web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -108,19 +117,25 @@ export class LayerDialogComponent implements OnDestroy {
     if (!this.projectId) {
       throw Error('Project not yet loaded');
     }
+    let fields = Map<string, Field>();
+    this.layerForm.value.questions.map((question: { en: string }) => {
+      const fieldId = this.dataStoreService.generateId();
+      const field: Field = {
+        id: fieldId,
+        type: FieldType['TEXT'],
+        required: false,
+        label: StringMap(question),
+      };
+      fields = fields.set(fieldId, field);
+    });
     const formId = this.dataStoreService.generateId();
-    const fieldId = this.dataStoreService.generateId();
     const layer = new Layer(
       this.layerId,
-      this.layer?.color,
+      this.layer?.color ? this.layer.color : '#ff9131',
       this.layer?.name?.set(this.lang, this.layerName),
-      this.layerForm.value.question
+      this.layerForm.value.questions
```

Comment:
Is `questions` undefined when there are no items, or empty array?

Reply:
Question s till stands - what happens if questions is empty?

Reply:
questions is not empty it is initialized with  { 'en': '' }

Reply:
I'm probably confused about how layerForm works - what is this.layerForm.value.questions? From its now I thought it was an array of components?

Reply:
Also, why does this ternary use fields but check `questions`? Also, can't `this.getForm` return the right thing directly instead of checking it here?

Reply:
We spoke - `questions` is never null, but in theory can be empty. we should handle those cases here instead.